go-controller/pkg/node/gateway_localnet.go

```diff
@@ -183,15 +193,22 @@ func (npw *localPortWatcherData) deleteService(svc *kapi.Service) error {
 				klog.V(5).Infof("ExternalIP: %s is reachable through one of the interfaces on this node, will skip cleanup", externalIP)
 			} else {
 				if gatewayIP != "" {
-					if stdout, stderr, err := util.RunIP("route", "del", externalIP, "via", gatewayIP, "dev", localnetGatewayNextHopPort, "table", localnetGatewayExternalIDTable); err != nil {
-						klog.Errorf("Error delete routing table entry for ExternalIP %s: stdout: %s, stderr: %s, err: %v", externalIP, stdout, stderr, err)
-					} else {
-						klog.V(5).Infof("Successfully deleted route for ExternalIP: %s", externalIP)
-					}
+					routeUsage[externalIP] = true
 				}
 			}
 		}
 	}
+
+	for externalIP, hasRoute := range routeUsage {
+		if hasRoute {
+			if stdout, stderr, err := util.RunIP("route", "del", externalIP, "via", gatewayIP, "dev", localGwModeNextHopPort, "table", localnetGatewayExternalIDTable); err != nil {
```

Comment:
netlink.RouteDel()

Reply:
done

Reply:
@dcbw I started down the path of converting all of this to use netlink. It ended up being a lot more work than I initially thought:
https://github.com/trozet/ovn-kubernetes/commit/23bf2acf863b135a4c7da044ee052bb0d18c7cfe

and I still haven't mocked out the ginkgo tests or added unit tests yet. I'd prefer if we can keep that as a separate PR so we can get this in faster.